sdk/go/x/auto/local_workspace.go

```diff
@@ -409,6 +411,62 @@ func (l *LocalWorkspace) SetProgram(fn pulumi.RunFunc) {
 	l.program = fn
 }
 
+// ExportStack exports the deployment state of the stack matching the given name.
+// This can be combined with ImportStack to edit a stack's state (such as recovery from failed deployments).
+func (l *LocalWorkspace) ExportStack(ctx context.Context, stackName string) (apitype.UntypedDeployment, error) {
+	var state apitype.UntypedDeployment
+	err := l.SelectStack(ctx, stackName)
+	if err != nil {
+		return state, errors.Wrapf(err, "could not export stack, unable to select stack %s.", stackName)
+	}
+
+	stdout, stderr, errCode, err := l.runPulumiCmdSync(ctx, "stack", "export", "--show-secrets")
+	if err != nil {
+		return state, newAutoError(errors.Wrap(err, "could not export stack."), stdout, stderr, errCode)
+	}
+
+	err = json.Unmarshal([]byte(stdout), &state)
+	if err != nil {
+		return state, newAutoError(
+			errors.Wrap(err, "failed to export stack, unable to unmarshall stack state."), stdout, stderr, errCode,
+		)
+	}
+
+	return state, nil
+}
+
+// ImportStack imports the specified deployment state into a pre-existing stack.
+// This can be combined with ExportStack to edit a stack's state (such as recovery from failed deployments).
+func (l *LocalWorkspace) ImportStack(ctx context.Context, stackName string, state apitype.UntypedDeployment) error {
+	err := l.SelectStack(ctx, stackName)
+	if err != nil {
+		return errors.Wrapf(err, "could not import stack, failed to select stack %s.", stackName)
+	}
+
+	f, err := ioutil.TempFile(os.TempDir(), "")
+	if err != nil {
+		return errors.Wrap(err, "could not import stack. failed to get allocate temp file.")
+	}
+	defer func() { contract.IgnoreError(os.Remove(f.Name())) }()
+
+	bytes, err := json.Marshal(state)
+	if err != nil {
+		return errors.Wrap(err, "could not import stack, failed to marshal stack state.")
+	}
+
+	_, err = f.Write(bytes)
+	if err != nil {
+		return errors.Wrap(err, "could not import stack. failed to write out stack intermediate.")
+	}
+
```

Comment:
```suggestion
// Close the file
if err := f.Close(); err != nil {
  return errors.Wrap(err, "could not import stack. failed to close out stack intermediate")
}
```

Reply:
I believe this is handled by the defer a few lines up https://github.com/pulumi/pulumi/pull/5369/files#diff-226d1106206f1bf744142667c0ade384R450